### DISCUSSION TITLE
How can I destroy and init again an existing swiper using some changed / added parameters?

### DISCUSSION THREAD

**Discussion by martin-petzold** [nolimits4web/swiper#7497] [Q&A]
I have an existing swiper instance and would like to change / add some additional parameters after first initialization.

However, I can't get it working. This does not work:

```
var parameters = swiper.originalParams();
parameters['slidesPerView'] = 4;
swiper.destroy(false, false);
swiper.init(parameters);
```
OR
```
var parameters = swiper.params();
parameters['slidesPerView'] = 4;
swiper.destroy(false, false);
swiper.init(parameters);
```
OR
```
var parameters = swiper.originalParams();
parameters['slidesPerView'] = 4;
swiper.destroy(false, true);
swiper.init(parameters);
```
OR
```
var parameters = swiper.params();
parameters['slidesPerView'] = 4;
swiper.destroy(false, true);
swiper.init(parameters);
```
After this I don't have four slides on the view (still one) and also the navigation does not work anymore.

Thanks,

Martin

**Comment by martin-petzold** [comment]
I checked 'params' before and after init. Both are identical, and 'slidesPerView' is changed to 4.

**Comment by TalantRys** [comment]
Omg, a whole year passed...
Params actually are objects, you have to change params like this:
```
const slider = new Swiper();
slider.params['slidesPerView'] = 4; // or slider.params.slidesPerView = 4;
swiper.update();
```

But it doesn't work well with breakpoints. I opened my discussion about that: #7949
